doc/subsystems/networking/bsd-sockets.rst

```diff
@@ -0,0 +1,65 @@
+.. _bsd_sockets_api:
+
+BSD Sockets compatible API
+##########################
+
+Zephyr offers an implementation of a subset of BSD Sockets API (a part
+of POSIX standard). This allows to reuse existing programming experience
+and port existing simple networking applications to Zephyr.
+
+Following are the requirements and concepts which governed BSD Sockets
+compatible API implementation for Zephyr:
+
+* Should have minimal overhead, similar to the requirement for other
+  Zephyr subsystems.
+* Should be implemented on top of
+  :ref:`native networking API <networking_api_usage>` to keep modular
+  design.
+* Should be namespaced by default, to avoid name conflicts with well-known
+  names like ``close()`` which may be part of libc or other POSIX
+  compatibility libraries. If enabled, should also expose native POSIX
+  names.
+
+BSD Sockets compatible API is enabled using :option:`CONFIG_NET_SOCKETS`
+config option and implements following operations: ``socket()``,
+``close()``, ``recv()``, ``send()``, ``connect()``, ``bind()``,
+``listen()``, ``fcntl()`` (to set non-blocking mode), ``poll()``.
+
+Based on the namespacing requirements above, these operations are by
+default exposed as functions with ``zsock_`` prefix, e.g.
+:c:func:`zsock_socket()` and :c:func:`zsock_close()`. If config option
+:option:`CONFIG_NET_SOCKETS_POSIX_NAMES` is defined, all the functions
+will be also exposed as aliases without the prefix. This includes the
+functions like ``close()`` and ``fcntl()`` (which may conflict with
+functions in libc or other libraries (e.g. with filesystem libraries)).
+
+Native BSD Sockets API uses file descriptors to represent sockets,
```

Comment:
I'm pasting this as is, but this finishing "instead of using ..., use ..." leaves an impression there's some choice on user side. There's none - it's a design decision to provide poll() and not provide select().